### PR TITLE
[Merged by Bors] - chore: Let `gcongr` know about `Nat.cast_lt`

### PR DESCRIPTION
**Pull request by YaelDillies**
From LeanAPAP

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Nat/Cast/Order/Basic.lean [hunk 1]
```diff
@@ -69,6 +69,9 @@ variable [CharZero α] {m n : ℕ}
 theorem strictMono_cast : StrictMono (Nat.cast : ℕ → α) :=
   mono_cast.strictMono_of_injective cast_injective
 
+@[gcongr]
+lemma _root_.GCongr.natCast_lt_natCast {a b : ℕ} (h : a < b) : (a : α) < b := strictMono_cast h
```
Comment (j-loreaux) [inline Mathlib/Data/Nat/Cast/Order/Basic.lean:73]: Can we just make this an `alias`? And add a docstring to that effect?

**Review by j-loreaux — COMMENTED** [review]
bors d+

File: Mathlib/Data/Nat/Cast/Order/Basic.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Data/Nat/Cast/Order/Basic.lean:73]: An alias of what? `strictMono_cast`? `gcongr` doesn't understand `StrictMono`

**Comment by YaelDillies** [comment]
I think your remark isn't crucial, Jireh, so I will march on towards the juicy stuff in APAP.

bors merge

**Pull request closed without merging**